### PR TITLE
Tutorial: Move QL detective tutorial library into shared `codeql/tutorial` library pack

### PR DESCRIPTION
**Pull request by adityasharad**
Key changes:
- Introduce a new shared pack `codeql/tutorial`, containing the `tutorial.qll` library used for the QL detective tutorials.
- All language library packs `codeql/*-all` now depend on `codeql/tutorial`. This means that `import tutorial` will work in any query that depends on any of the language libraries. Remove the duplicate copies of `tutorial.qll`.
  - For Go, Ruby, and Swift, this is new behaviour.
  - For other languages, this is existing behaviour.
- Adjust the QLDoc comments within the `tutorial` module. The `Person` class gets a new doc comment, and the `tutorial` module gets the existing doc comment that was previously attached to `Person`.

Not in scope in this PR:
- Changing the import path. This remains `import tutorial`.
- Depending on a specific dbscheme. The tutorial library right now has no extractor/dbscheme dependency, and can only be used from a query that does have a transitive dependency on some dbscheme.

@github/code-scanning-secexp-reviewers this should make it easier in future for us to reference the tutorial libraries from a Codespace, but it does not block the existing Codespaces work.

### CODE REVIEW

**Comment by adityasharad** [comment]
Apologies for the wide set of review requests! This touches all language libraries but I hope the changes are focused enough to review for your language separately.

**Review by erik-krogh — APPROVED** [review]
It all looks good 👍  I checked all the changes.  

At first I thought that you hadn't deleted `cpp/ql/lib/tutorial.qll`, but that's because it's been renamed to `shared/tutorial/tutorial.qll` according to `git`.

File: cpp/ql/lib/qlpack.yml [hunk 1]
```diff
@@ -7,3 +7,4 @@ library: true
 upgrades: upgrades
 dependencies:
   codeql/ssa: ${workspace}
+  codeql/tutorial: ${workspace}
```
Comment (aibaars) [inline cpp/ql/lib/qlpack.yml:10]: What is the reason for adding a dependency on the tutorial pack? The pack is unused.

Comment (erik-krogh) [reply cpp/ql/lib/qlpack.yml:10]: Such that you can do `import tutorial` from any file.

**Review by jketema — APPROVED** [review]

**Review by owen-mc — APPROVED** [review]
Approved for Go

**Pull request merged**